src/App.js

```diff
@@ -14,11 +14,76 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state ={
+    choosenGoods: [],
+    isButtonForDeleteAllVisible: false,
+  }
+
+  addGood = (good) => {
+    if (this.state.choosenGoods.includes(good)) {
+      this.setState(prevState => (
+        { choosenGoods: prevState.choosenGoods.filter(el => el !== good) }
+      ));
+    } else {
```

Comment:
```suggestion
    if (this.state.choosenGoods.includes(good)) {
      this.setState(prevState => (
        { choosenGoods: prevState.choosenGoods.filter(el => el !== good) }
      ));
    } else {
```

спробуй деструктуризацію, якщо хочеш)

`el?`

Reply:
What is el ?